internal/replacements/replacements_test.go

```diff
@@ -290,12 +290,69 @@ var _ = Describe("replace_misconfigured_pods", func() {
 				pod.Spec.NodeSelector = map[string]string{
 					"dummy": "test",
 				}
-				needsRemoval, err := processGroupNeedsRemovalForPod(cluster, pod, processGroup, log)
+				needsRemoval, err := processGroupNeedsRemovalForPod(cluster, pod, processGroup, log, true)
 				Expect(needsRemoval).To(BeFalse())
 				Expect(err).NotTo(HaveOccurred())
 			})
 		})
 
+		Context("when the securityContext doesn't match", func() {
+			BeforeEach(func() {
+				pClass = fdbv1beta2.ProcessClassStorage
+				remove = false
+			})
+
+			It("should need a removal for RunAsUser change", func() {
+				pod.ObjectMeta.Annotations[fdbv1beta2.LastSpecKey] = "banana"
+				// if ReplaceInstancesWhenResourcesChange is true, any spec change should result in replacement
+				cluster.Spec.ReplaceInstancesWhenResourcesChange = new(bool)
+
+				pod.Spec.SecurityContext = &corev1.PodSecurityContext{RunAsUser: new(int64)}
+				needsRemoval, err := processGroupNeedsRemovalForPod(cluster, pod, processGroup, log, true)
+				Expect(needsRemoval).To(BeTrue())
+				Expect(err).NotTo(HaveOccurred())
+			})
+			It("should need a removal for FSGroup change", func() {
+				pod.ObjectMeta.Annotations[fdbv1beta2.LastSpecKey] = "banana"
+				// if ReplaceInstancesWhenResourcesChange is true, any spec change should result in replacement
+				cluster.Spec.ReplaceInstancesWhenResourcesChange = new(bool)
+
+				pod.Spec.SecurityContext = &corev1.PodSecurityContext{FSGroup: new(int64)}
+				needsRemoval, err := processGroupNeedsRemovalForPod(cluster, pod, processGroup, log, true)
+				Expect(needsRemoval).To(BeTrue())
+				Expect(err).NotTo(HaveOccurred())
+			})
+			It("with no spec change, it should not need a removal for FSGroup change to guard against server-side defaults", func() {
+				// do not change last spec key
+				// if ReplaceInstancesWhenResourcesChange is true, any spec change should result in replacement
+				cluster.Spec.ReplaceInstancesWhenResourcesChange = new(bool)
+
+				pod.Spec.SecurityContext = &corev1.PodSecurityContext{FSGroup: new(int64)}
+				needsRemoval, err := processGroupNeedsRemovalForPod(cluster, pod, processGroup, log, true)
+				Expect(needsRemoval).To(BeFalse())
+				Expect(err).NotTo(HaveOccurred())
+			})
+			It("should need a removal with ReplaceInstancesWhenResourcesChange", func() {
```

Comment:
Wrong text in the `It` statement? How is the `ReplaceInstancesWhenResourcesChange` related to this test case?